mapstructure.go

```diff
@@ -451,6 +455,8 @@ func (d *Decoder) decode(name string, input interface{}, outVal reflect.Value) e
 		}
 	}
 
+	performNilDecoding := d.config.DecodeNil && d.config.DecodeHook != nil
```

Comment:
```suggestion
	decodeNil := d.config.DecodeNil && d.config.DecodeHook != nil
```
use same name, less cognitive overhead